AnalysisNtuple/makeAnalysisNtuple.C

```diff
@@ -702,13 +725,70 @@ makeAnalysisNtuple::makeAnalysisNtuple(int ac, char** av)
 	    }
 	}
 
+	if( isMC && doOverlapInvert_GJ){
+	    //if (!overlapRemoval_Tchannel(tree)){
+	    if (!overlapRemoval(tree, 25., 2.5, 0.4, tree->event_==eventNum)){
+		count_overlap++;
+		continue;
+	    }
+	}
+
+	if( isMC && doOverlapRemoval_QCD){
+	    if (overlapRemoval(tree, 25., 2.5, 0.4, tree->event_==eventNum)){
+		count_overlap++;
+		continue;
+	    }
+	}
+
 	// //		Apply systematics shifts where needed
 	if( isMC ){
 	    if (jecvar012_g != 1){
 		jecvar->applyJEC(tree, jecvar012_g); // 0:down, 1:norm, 2:up
 	    }
 	}
 
+	//HEM test 
+	int nHEM_ele=0;
+	bool HEM_ele_Veto=false;
+    	for(int eleInd = 0; eleInd < tree->nEle_; ++eleInd){
+        	double eta = tree->eleEta_[eleInd];
+ 		double pt = tree->elePt_[eleInd];
+        	double phi = tree->elePhi_[eleInd];
+		bool ele_HEM_pt_pass  = pt >= 15 ;
+        	bool ele_HEM_eta_pass = eta > -3.0 && eta < -1.4 ;
+        	bool ele_HEM_phi_pass = phi > -1.57 && phi < -0.87;
+        	if ( ele_HEM_pt_pass &&  ele_HEM_eta_pass &&  ele_HEM_phi_pass) nHEM_ele++;
+	}
+  	if (nHEM_ele>=1) {
+        	HEM_ele_Veto=true;
+        }
```

Comment:
Better way to do this is to set HEM_ele_Veto to the conditional statement, so that when the condition is false it is explicitly set to false.
`HEM_ele_Veto=(nHEM_ele>=1)`